core/readme/README.md

```diff
@@ -17,5 +17,28 @@
 npm install @verdaccio/readme
 ```
 
+## Usage
+
+Simple usage
+
+```js
+import parseReadme from '@verdaccio/readme';
+
+const markdownString = '## hello world';
+
+parseReadme(markdownString);
+```
+
+Using syntax highlighter
+
+```js
+import parseReadme from '@verdaccio/readme';
+
+const markdownString = '## hello world';
+
+parseReadme(markdownString, true); // true for highlighting syntax
```

Comment:
I think we should use an `options` object.

Reply:
yeah...that makes more sense!!!
should I add all [these options](https://marked.js.org/#/USING_ADVANCED.md#options) support from marked ?

Reply:
No, just pass it through as hlOptions.

Reply:
```js
type hlOptionsType {
enable : boolean = false
}
export default function parseReadme(readme: string, hlOptions : hlOptionsType): string | void {
  if (hlOptions.enable) {
    marked.setOptions({
      highlight: function(code, lang) {
        return hljs.highlight(lang, code).value;
      },
    });
  }

```


Like this ?

Reply:
Not what I thought. I thought we directly pass the options through and set the `highlight` function.

What you use is a boolean which does not accept more settings.

Reply:
```js
mdOptions = {
  highlight: function(code, lang) {
    return hljs.highlight(lang, code).value;
  }
};
```

and using this !!

```js
type mdOptionsType {
highlight : Function,
// ...anything else ?
}

export default function parseReadme(readme: string, Options:mdOptionsType ): string | void {
  marked.setOptions(Options);
}
```

and calling this using

```js
parseReadme(string, mdOptions);
```

Reply:
cc @DanielRuf

Reply:
cc @DanielRuf reminder

Reply:
Still not the solution that I would recommend, pass through a `hlOptions`variable and in the method use `Object.assign({}, {highlight: ...}, hlOptions)`.

The type is too strict (we do not and can not even define the shape, it may change) and the other solution does not cover more custom options which we may want to provide / set.

Reply:
> `Object.assign({}, {highlight: ...}, hlOptions).`

I think this may work

```typescript
export default function parseReadme(
  readme: string,
  options: mdOptionsType
): string | void {
  mdoptions = Object.assign({}, { ...some_of_our_options }, options);
  marked.setOptions(mdoptions);
}
```

?

Reply:
This looks much better, yes.

Reply:
This looks much better, yes.

Reply:
Yes, this looks much better.

Reply:
ok